system-tests/projects/e2e/cypress/integration/no_pause_spec.js

```diff
@@ -0,0 +1,27 @@
+describe('cy.pause()', () => {
+  it('pauses', () => {
+    let didPause = false
+
+    cy.visit('https://example.cypress.io')
+
+    cy.once('paused', (name) => {
+      cy.once('paused', (name) => {
+        didPause = true
+
+        // resume the rest of the commands so this
+        // test ends
+        Cypress.emit('run:end')
+        Cypress.emit('resume:all')
+      })
+
+      Cypress.emit('resume:next')
+      Cypress.emit('run:end')
+    })
+
+    cy.pause().wrap({}).should('deep.eq', {}).then(function () {
+      expect(didPause).to.be.false
```

Comment:
personally, I would combine `no_pause_spec` and `pause_spec` and switch on behavior with `Cypress.browser.isHeadless` or by passing `env` in, but since it's a test it's fine to just copy it too

Reply:
Wouldn't I need to factor in `noExit` as well to switch on behavior? Since if `--headed` was passed but `--no-exit` was not, we _do not_ want to pause?